lib/setup_functions.php

```diff
@@ -24,14 +24,15 @@ function checkModulePresence()
     $sSQL .= $sModulePath . "/nominatim.so', 'transliteration' LANGUAGE c IMMUTABLE STRICT";
     $sSQL .= ';DROP FUNCTION nominatim_test_import_func(text);';
 
-    $oDB = &getDB();
-    $oResult = $oDB->query($sSQL);
+    $oDB = new \Nominatim\DB();
+    $oDB->connect();
+    $oResult = $oDB->exec($sSQL);
 
     $bResult = true;
 
-    if (PEAR::isError($oResult)) {
+    if ($oResult === false) {
```

Comment:
Here too. I'd expect exec() to throw.

Reply:
My point was that the whole of the if statement is never reached. You need to try/catch the exec.

I've confirmed that in the meantime. Delete module/nominatim.so and watch an uncaught exception being thrown. (BTW, the exception is not very informative as the stack trace is missing for some reason, so it just reports the rethrow as a source. We might need exception handlers in init-cmd.php too.)